datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Domain/SetDomainModal.tsx

```diff
@@ -181,40 +181,42 @@ export const SetDomainModal = ({ urns, onCloseModal, refetch, defaultValue, onOk
-                            placeholder="Search for Domains..."
-                            onSelect={(domainUrn: any) => onSelectDomain(domainUrn)}
-                            onDeselect={onDeselectDomain}
-                            onSearch={(value: string) => {
-                                // eslint-disable-next-line react/prop-types
-                                handleSearch(value.trim());
-                                // eslint-disable-next-line react/prop-types
-                                setInputValue(value.trim());
-                            }}
-                            ref={inputEl}
-                            value={selectValue}
-                            tagRender={tagRender}
-                            onBlur={handleBlur}
-                            onFocus={() => setIsFocusedOnInput(true)}
-                            dropdownStyle={isShowingDomainNavigator ? { display: 'none' } : {}}
-                        >
-                            {domainSearchOptions}
-                        </Select>
-                        <BrowserWrapper isHidden={!isShowingDomainNavigator}>
-                            <DomainNavigator selectDomainOverride={selectDomainFromBrowser} />
-                        </BrowserWrapper>
-                    </ClickOutside>
-                </Form.Item>
-            </Form>
+                <Form component={false}>
+                    <Form.Item>
+                        <ClickOutside onClickOutside={handleCLickOutside}>
+                            <Select
+                                autoFocus
+                                defaultOpen
+                                filterOption={false}
+                                showSearch
+                                mode="multiple"
+                                defaultActiveFirstOption={false}
+                                placeholder="Search for Domains..."
+                                onSelect={(domainUrn: any) => onSelectDomain(domainUrn)}
+                                onDeselect={onDeselectDomain}
+                                onSearch={(value: string) => {
+                                    // eslint-disable-next-line react/prop-types
+                                    handleSearch(value.trim());
+                                    // eslint-disable-next-line react/prop-types
+                                    setInputValue(value.trim());
+                                }}
+                                ref={inputEl}
+                                value={selectValue}
+                                tagRender={tagRender}
+                                onBlur={handleBlur}
+                                onFocus={() => setIsFocusedOnInput(true)}
+                                dropdownStyle={isShowingDomainNavigator ? { display: 'none' } : {}}
+                                loading={loading}
+                                disabled={loading}
```

Comment:
should we definitely disable this if it's not done loading yet? for example, if they want to search for a specific domain, they should probably be able to do that right away?

not blocking

Reply:
As soon as the user opens the modal, we kick off an API call to fetch the Domains. Since the user won't be typing anything initially, it can be a bit confusing to see an empty list when they first focus on the search field. 

To avoid this confusion, we disable the search field and show a loader to signal that we're in the process of loading the initial list. Once the domains are ready, we enable the search functionality, allowing the user to start their search.

 Even while they're searching, if they momentarily pause typing, we have a timer in place to trigger a search API call. During this time, the search field remains disabled with a loader, indicating that we're actively fetching results that match their input.

Reply:
This should be resolved now. No longer any disabled state, as discussed during our standup